src/test/java/com/hocs/test/glue/DataInputStepDefs.java

```diff
@@ -200,4 +203,78 @@ public void assertValidationMessagesOnDataInputForm(String field) {
                 pendingStep(field + " is not defined within " + getMethodName());
         }
     }
+
+    @Then("^the correspondence type is the \"([^\"]*)\" correspondent$")
+    public void theCorrespondenceTypeIsTheCorrespondent(String ordinal) {
+        switch (ordinal.toUpperCase()) {
+            case "PRIMARY":
+                recordCorrespondentDetails.assertPrimaryCorrespondent();
+                break;
+            case "SECONDARY":
+                recordCorrespondentDetails.assertSecondaryCorrespondent();
+                break;
+            default:
+                pendingStep(ordinal + " is not defined within " + getMethodName());
+        }
+    }
+
+    @And("^a case has a \"([^\"]*)\" correspondent$")
+    public void aCaseHasACorrespondent(String ordinal) {
+        switch (ordinal.toUpperCase()) {
+            case "PRIMARY":
+                setSessionVariable("caseReference").to(getHeaderText().substring(10));
+                dataInput.fillAllMandatoryCorrespondenceFields();
+                genericInputStepDefs.clickTheButton("Continue");
+                addACorrespondentThatIsOrIsNotAnMP("Is not");
+                recordCorrespondentDetails.fillMandatoryCorrespondentFields();
+                dataInput.clickAddButton();
```

Comment:
As above `clickOn();`

Reply:
amended underlying clickAddButton() method to use clickOn() instead of .click()